src/base/bittorrent/trackerentry.h

```diff
@@ -54,14 +54,14 @@ namespace BitTorrent
         TrackerEntry &operator=(const TrackerEntry &other) = default;
 
         QString url() const;
-        Status status() const;
+        Status status(InfoHashFormat hashVersion = InfoHashFormat::V1) const;
```

Comment:
Does it make sense to call this method with different hash versions for the same torrent entry?
Its use is unclear to me, since you don't use it anywhere within this PR.
The same question is about all similar methods.

Reply:
>Its use is unclear to me, since you don't use it anywhere within this PR.

After we fully support bittorent v2, this default value would be removed and the caller would need to provide the intended parameter.

>Does it make sense to call this method with different hash versions for the same torrent entry?

I have no definite idea right now. I made it (`InfoHashFormat`) a function parameter instead of a class variable is because I reckon it doesn't make sense for this class (`TrackerEntry`) to specify a hash format when invoking the constructors, for example the default constructor and the constructor taking a QString.
This is loosely following the design of libtorrent which they always provide 2 info hashes despite sometimes only one is in-use, and to find out which is in-use one needs to query another class `torrent_handle::info_hashes().has()`.

Reply:
I'll try to ask the question in a different way: does it make sense to request data using a hash of a type other than tracker entry torrent has? I.e. if torrent has vX hash does it make sense to query its tracker entries for any info that associated with vY hash?

Reply:
I think it depends, if a torrent only has v1 hash then querying for v2 hash doesn't make sense. If a torrent has both v1, v2 hashes then it might be reasonable to know the status of each hash.

Reply:
>If a torrent has both v1, v2 hashes then it might be reasonable to know the status of each hash.

It can make sense.
>if a torrent only has v1 hash then querying for v2 hash doesn't make sense.

Then how do you intend to handle it in case of misuse?
I think the caller should be responsible for this (i.e. checking the available hash version before requesting the related information). But there should still be some sort of handling of misuse. It should at least return an adequate response (for example, if info related to v2 hash is requested for a torrent that has only v1 hash, then it should return 0 peers, 0 seeds, etc., and some appropriate tracker status). How does it behave now?
And one more remark. It seems that a request for Both hash versions for a torrent that has only one should still be regarded as a normal situation (i.e. if the torrent has only v1hash, then the request for both v1 and Both should return the same result), do you agree?

Reply:
Or maybe return `std::optional` from all similar methods so you can return `std::nullopt` in case when unsupported hash format is requested?

Reply:
>It should at least return an adequate response (for example, if info related to v2 hash is requested for a torrent that has only v1 hash, then it should return 0 peers, 0 seeds, etc., and some appropriate tracker status). 

I agree that it should return `0`.

>And one more remark. It seems that a request for Both hash versions for a torrent that has only one should still be regarded as a normal situation (i.e. if the torrent has only v1hash, then the request for both v1 and Both should return the same result), do you agree?

Perhaps it is a mistake for this function to accept `Both` option. This function should accept `V1`, `V2` in form of flags (or QFlags).